### PR TITLE
feat(fs): initial fs directory APIs

### PR DESCRIPTION
**Pull request by jwerle**

### CODE REVIEW

File: test/mock.js [hunk 1]
```diff
@@ -163,7 +163,8 @@ global.window = Object.assign(new EventEmitter(), {
   },
 
   process: {
-    index: 0
+    index: 0,
+    platformn: process.platform
```
Comment (chicoxyzzy) [inline test/mock.js]:
```suggestion
    platform: process.platform
```

File: util.js [hunk 2]
```diff
@@ -59,15 +67,25 @@ export function InvertedPromise () {
...
+        return promise
       }
     })
   })
 
   return Object.assign(promise, context)
 }
+
+export function clamp (value, min, max) {
+  if (Number.isNaN(value) || typeof value !== 'number' || !Number.isFinite(value)) {
```
Comment (chicoxyzzy) [inline util.js]: `Number.isFinite` handles `NaN`s and checks that argument is number literal or `Number` instance, so other conditions are not necessary here

File: fs/index.js [hunk 3]
```diff
@@ -344,9 +362,53 @@ export function read (fd, buffer, offset, length, position, options, callback) {
 }
 
 /**
- * @TODO
+ * Asynchronously read all entries in a directory.
+ * @see {https://nodejs.org/dist/latest-v16.x/docs/api/fs.html#fsreaddirpath-options-callback}
+ * @param {string | Buffer | URL | number } path
```
Comment (chicoxyzzy) [inline fs/index.js]: couldn't be a number

File: fs/dir.js [hunk 4]
```diff
@@ -1,56 +1,273 @@
...
   /**
-   * @TODO
+   * `Dir` class constructor.
+   * @param {DirectoryHandle} handle
+   * @param {?(object)} options
    */
-  constructor (options) {
-    this.path = options?.path || null
+  constructor (handle, options) {
+    this.path = handle?.path || null
```
Comment (chicoxyzzy) [inline fs/dir.js]: maybe `??` instead of `||`will make more sense, but I think that `''` is invalid path anyway?

**Review by chicoxyzzy — APPROVED** [review]

File: util.js (hunk 2 above)
Comment (jwerle) [reply util.js]: nice

**Pull request merged**